client/asset/interface.go

```diff
@@ -1765,3 +1765,10 @@ func (e *WalletEmitter) ActionResolved(uniqueID string) {
 		UniqueID: uniqueID,
 	})
 }
+
+// InitChecker is an account wallet that checks the status of a swap init.
+type InitChecker interface {
+	// AlreadyInitialized returns if the swap was already initialized and
+	// some data needed to interact with the contract later.
+	AlreadyInitialized(assetVersion uint32, contract *Contract) (initialized bool, zeroHash, contractData []byte, err error)
+}
```

Comment:
Could `Swap` just check whether or not the swap is already initialized, the same way how we check `isRedeemable` and `isRefundable`, and if it's already initialized then return without error as if the transaction was just sent. Then this interface and the `checkInitialized` function in `trade.go` would not be needed. The server just needs to be updated to not require the transaction hash.

Reply:
I didn't want to update the server as thats a bigger change. But we can do that. May not have time to work on this until week after next.

Reply:
I'll take a look.